From 84fe908e64a6e904166f728860ab8dfaa47850d2 Mon Sep 17 00:00:00 2001
From: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
Date: Fri, 6 Mar 2020 10:28:09 +0000
Subject: [PATCH 2/2] rfc3315: fix incorrect logical '&&' warning

rfc3315.c:1711:28: warning: use of logical '&&' with constant operand [-Wconstant-logical-operand]
    if (!(addr_list->flags && ADDRLIST_DECLINED) ||
                           ^  ~~~~~~~~~~~~~~~~~

It's a flag bit so should be bitwise '&' operator

Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
---
 src/rfc3315.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/rfc3315.c
+++ b/src/rfc3315.c
@@ -1708,7 +1708,7 @@ static int config_valid(struct dhcp_conf
     return 0;
 
   for (addr_list = config->addr6; addr_list; addr_list = addr_list->next)
-    if (!(addr_list->flags && ADDRLIST_DECLINED) ||
+    if (!(addr_list->flags & ADDRLIST_DECLINED) ||
 	difftime(now, addr_list->decline_time) >= (float)DECLINE_BACKOFF)
       {
 	addrpart = addr6part(&addr_list->addr.addr6);
